flipperdevices/flipperzero-firmware
# [FL-3123] SD Cache: moved to diskio layer, invalidation in case of error

# What's new

- Simpler version of #2418 
- Moved SD cache to diskio level, no longer need to block spi bus if cache hit
- In case of a read error, the cache will be invalidated

# Verification 

- Everything that works with the sd card is affected
- Use the Flipper as you would normally, check for any abnormal behavior
- Remove/swap/reconnect the SD card in the process of using the Flipper and check that it doesn't slow down or crash

# Checklist (For Reviewer)
